src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -58,15 +58,15 @@ void eicrecon::TrackPropagation_factory::Process(const std::shared_ptr<const JEv
 
     for(auto traj: trajectories) {
         edm4eic::MutableTrackSegment this_propagated_track;
-        for(size_t isurf = 0; auto surf: m_target_surface_list) {
+        for(size_t isurf = 0; isurf < m_target_surface_list.size(); isurf++) {
+            auto surf = m_target_surface_list[isurf];
             auto prop_point = m_track_propagation_algo.propagate(traj, surf);
             if(!prop_point) continue;
 #if EDM4EIC_VERSION_MAJOR >= 3
             prop_point->surface = m_target_surface_ID[isurf];
             prop_point->system = m_target_detector_ID[isurf];
```

Comment:
I think this should work, and avoids the need for the index or even the variables that store the surface and detector ID:
```suggestion
            prop_point->surface = surf.geometryId().layer();
            prop_point->system = surf.geometryId().extra();
```